src/annotator/integrations/vitalsource.js

```diff
@@ -55,12 +55,36 @@ export class VitalSourceContainerIntegration {
       throw new Error('Book container element not found');
     }
 
+    /** @type {WeakSet<HTMLIFrameElement>} */
+    const contentFrames = new WeakSet();
```

Comment:
I believe this is to track if the Hypothesis client has already been injected into this iframe's document. However, it stores the iframe element.

Is re-injection of the client possible?

I think this check is unnecessary if `hasHypothesis` function worked.

Reply:
> I believe this is to track if the Hypothesis client has already been injected into this iframe's document. 

This is used to track whether we have set up the `load` event listener on the frame.